playbooks/common/openshift-node/config.yml

```diff
@@ -84,6 +87,7 @@
     args:
       creates: "{{ etcd_generated_certs_dir }}/{{ item.etcd_cert_subdir }}.tgz"
     with_items: etcd_needing_client_certs
+    when: etcd_client_flannel_certs_missing is defined and etcd_client_flannel_certs_missing
```

Comment:
Also, we could probably just have a second set_fact above for the 'etcd_client_flannel_certs_missing is not defined' that would set etcd_needing_client_certs to []

Might be a better quick workaround until we can refactor this up a bit.